### PR TITLE
Fixes #24966 - Move mongo upgrade inside upgrade if logic.

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

**Comment by ehelms** [comment]
@sean797 This is targeted at your upgrade issue -- if you want to review it

File: hooks/post/30-upgrade.rb [hunk 1]
```diff
@@ -59,11 +59,10 @@ def fail_and_exit(message)
   kafo.class.exit 1
 end
 
-upgrade_step :remove_legacy_mongo
-
 if app_value(:upgrade)
   if [0, 2].include?(@kafo.exit_code)
     upgrade_tasks if module_enabled?('foreman')
+    remove_legacy_mongo if module_enabled?('foreman')
```
Comment (ekohl) [inline hooks/post/30-upgrade.rb]: I'd expect this to be done when the `katello` or `foreman_proxy_content` module is enabled.

Comment (ekohl) [reply hooks/post/30-upgrade.rb]: You're also removing the `upgrade_step`. Is this intentional?

**Comment by chris1984** [comment]
@ekohl updated, here are the results of the change:

Before:
```
  Proxy:

Installing             Done                                               [100%] [................................................................................................................................................................................................]
  Success!
  * Foreman Proxy is running at https://cap.katello.lan:9090
  The full log is at /var/log/foreman-installer/foreman-proxy-content.log
  Upgrade Step: remove_legacy_mongo...
  yum install -y -q rh-mongodb34-syspaths finished successfully!

  Katello:
  
  Success!
  * Katello is running at https://test.katello.lan
      Initial credentials are admin / bdFSAWtNAqjCNydn
  * To install an additional Foreman proxy on separate machine continue by running:

      foreman-proxy-certs-generate --foreman-proxy-fqdn "$FOREMAN_PROXY" --certs-tar "/root/$FOREMAN_PROXY-certs.tar"
	  The full log is at /var/log/foreman-installer/katello.log
  Upgrade Step: remove_legacy_mongo...
  [ INFO 2018-09-19T12:31:34 verbose] Upgrade Step: remove_legacy_mongo...
  [ INFO 2018-09-19T12:31:34 verbose] MongoDB 2.x not detected, skipping
  yum install -y -q rh-mongodb34-syspaths finished successfully!
  [ INFO 2018-09-19T12:31:37 verbose] All hooks in group post finished
  [ INFO 2018-09-19T12:31:37 verbose] Installer finished in 475.482776414 seconds
```
After:
```
  Katello:

  [ INFO 2018-09-19T11:54:57 verbose] Executing hooks in group post
  Success!
  * Katello is running at https://test.katello.lan
      Initial credentials are admin / JtuHtsGkGwZAoucW
  * To install an additional Foreman proxy on separate machine continue by running:

      foreman-proxy-certs-generate --foreman-proxy-fqdn "$FOREMAN_PROXY" --certs-tar "/root/$FOREMAN_PROXY-certs.tar"
  The full log is at /var/log/foreman-installer/katello.log
  [ INFO 2018-09-19T11:54:57 verbose] All hooks in group post finished

Upgrade:

  [root@test ~]# foreman-installer -v -s --disable-system-checks --scenario katello --upgrade
  Upgrade Step: remove_legacy_mongo...
  [ INFO 2018-09-19T12:04:49 verbose] Upgrade Step: remove_legacy_mongo...
  [ INFO 2018-09-19T12:04:49 verbose] MongoDB 2.x not detected, skipping
  Warning: RPMDB altered outside of yum.
  yum install -y -q rh-mongodb34-syspaths finished successfully!
  Upgrade completed!
  [ INFO 2018-09-19T12:04:52 verbose] Upgrade completed!
  [ INFO 2018-09-19T12:04:52 verbose] All hooks in group post finished
  [ INFO 2018-09-19T12:04:52 verbose] Installer finished in 284.605400811 seconds

  Capsule:
  
    Success!
  * Foreman Proxy is running at https://cap.katello.lan:9090
  The full log is at /var/log/foreman-installer/foreman-proxy-content.log
  [ INFO 2018-09-19T13:09:16 verbose] All hooks in group post finished
  [ INFO 2018-09-19T13:09:16 verbose] Installer finished in 66.219361232 seconds
```

**Pull request merged**